src/plugins/orangehrmRecruitmentPlugin/Api/CandidateAPI.php

```diff
@@ -406,13 +408,14 @@ private function setCandidate(Candidate $candidate): void
                 self::PARAMETER_COMMENT
             )
         );
+        $applicationDate = $this->getRequestParams()->getDateTimeOrNull(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_DATE_OF_APPLICATION,
+        );
         $candidate->setDateOfApplication(
-            $this->getRequestParams()->getDateTimeOrNull(
-                RequestParams::PARAM_TYPE_BODY,
-                self::PARAMETER_DATE_OF_APPLICATION,
-                null,
+            $applicationDate instanceof DateTime ?
+                $applicationDate :
                 $this->getDateTimeHelper()->getNow()
-            )
         );
         $candidate->setModeOfApplication(
```

Comment:
remove `self::PARAMETER_STATUS` and `self::PARAMETER_MODE_OF_APPLICATION` from validation rules and FE